scripts/cfFindIP.sh

```diff
@@ -79,18 +86,40 @@ if [ ! -d "$configDir" ]; then
     mkdir -p "$configDir"
 fi
 
+# Progress bar maker function
+# Base on https://www.baeldung.com/linux/command-line-progress-bar
+function show_progress {
+  current="$1"
+  total="$2"
+
+  bar_size="$(($(tput cols)-60))" # 65 cols for description characters
+
+  # calculate the progress in percentage 
+  percent=$(bc <<< "scale=$bar_percentage_scale; 100 * $current / $total" )
+  # The number of done and todo characters
+  done=$(bc <<< "scale=0; $bar_size * $percent / 100" )
+  todo=$(bc <<< "scale=0; $bar_size - $done")
+  # build the done and todo sub-bars
+  done_sub_bar=$(printf "%${done}s" | tr " " "${bar_char_done}")
+  todo_sub_bar=$(printf "%${todo}s" | tr " " "${bar_char_todo}")
+
+  # output the bar
+  progressBar="Progress : [${done_sub_bar}${todo_sub_bar}] ${percent}% of total IPs.         " # Some end space for pretty formatting
+}
+
 # Function fncCheckSubnet
 # Check Subnet
 function fncCheckSubnet {
 	local ipList scriptDir resultFile timeoutCommand domainFronting
 	ipList="$1"
-	resultFile="$2"
-	scriptDir="$3"
-	configId="$4"
-	configHost="$5"
-	configPath="$6"
-	configServerName="$7"
-	osVersion="$8"
+	resultFile="$3"
```

Comment:
Where is $2 ????

Reply:
The second argument is progress bar. I used Parallel's built-in progress bar (--bar) that show parameters values as a text in the progress bar. So first argument is IP and second is progress bar text and other arguments are followed.